WcaOnRails/app/controllers/teams_controller.rb

```diff
@@ -1,6 +1,8 @@
 class TeamsController < ApplicationController
   before_action :authenticate_user!
-  before_action -> { redirect_unless_user(:can_edit_teams?) }
+  before_action -> { redirect_unless_user(:can_manage_teams?) }, except: [:edit, :update]
+  before_action :assign_team, only: [:edit, :update]
+  before_action -> { redirect_unless_user(:can_edit_team?, @team) }, only: [:edit, :update]
```

Comment:
I would prefer to have a method `team_from_params` that looks up a team from the params, rather than this `assign_team`, but I personally find `before_action` rules hard to follow and kind of scary.